### PR TITLE
FOGL-1180 -  add syslog reader endpoint to REST api

### PR DESCRIPTION
**Pull request by amarendra-dianomic**
Facilitate  to see syslog entries via REST api (as we do for audit logs)

### CODE REVIEW

**Comment by praveen-garg** [comment]
can you please merge https://github.com/foglamp/FogLAMP/pull/638 into this? and then may be we just don't need syslogparser module/class at all and so the pyparsing dependency.

File: python/foglamp/services/core/api/support.py [hunk 1]
```diff
@@ -89,6 +91,42 @@
...
+    try:
+        cmd = "grep -n '{}\[' {} | tail -n {}".format(source, _SYSLOG_FILE, limit)
+        a = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE).stdout.readlines()
+        c = []
+        for b in a:
+            c.append(b.decode())
+    except (OSError, Exception) as ex:
+        raise web.HTTPException(reason=str(ex))
+
+    return web.json_response({'logs': c})
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/support.py]: please return count key with value too

File: python/foglamp/services/core/api/support.py [hunk 2]
```diff
@@ -89,6 +91,42 @@
     return web.json_response({"bundle created": bundle_name})
 
 
+async def get_syslog_entries(request):
+    """ Returns a list of syslog trail entries sorted with most recent first and total count
+        (including the criteria search if applied)
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/support.py]: default should return  latest 20 enteries

File: python/foglamp/services/core/api/support.py [hunk 3]
```diff
@@ -89,6 +91,42 @@
...
+
+    :Example:
+        curl -X GET http://localhost:8081/foglamp/syslog
+        curl -X GET "http://localhost:8081/foglamp/syslog?limit=5"
+        curl -X GET "http://localhost:8081/foglamp/syslog?source=Storage"
+        curl -X GET "http://localhost:8081/foglamp/syslog?limit=5&source=Storage"
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/support.py]: can we please make source case insensitive

File: python/foglamp/services/core/api/support.py [hunk 4]
```diff
@@ -89,6 +91,42 @@
...
+            raise ValueError
+    except ValueError:
+        raise web.HTTPBadRequest(reason="{} is not a valid source".format(source))
+
+    try:
+        cmd = "grep -n '{}\[' {} | tail -n {}".format(source, _SYSLOG_FILE, limit)
+        a = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE).stdout.readlines()
+        c = []
+        for b in a:
+            c.append(b.decode())
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/support.py]: so above 3 lines ` [c = b.decode() for b in a]` ? please add a comment what `b.decode does here;

File: python/foglamp/services/core/api/support.py [hunk 5]
```diff
@@ -20,6 +19,9 @@
 
 _FOGLAMP_DATA = os.getenv("FOGLAMP_DATA", default=None)
 _FOGLAMP_ROOT = os.getenv("FOGLAMP_ROOT", default='/usr/local/foglamp')
+_SYSLOG_FILE = '/var/log/syslog'
+__DEFAULT_LIMIT = 20
+__DEFAULT_OFFSET = 0
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/support.py]: no offset feature? however we defined this const

Comment (amarendra-dianomic) [reply python/foglamp/services/core/api/support.py]: removed

File: python/foglamp/services/core/api/support.py (hunk 2 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/core/api/support.py]: already there

File: python/foglamp/services/core/api/support.py (hunk 3 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/core/api/support.py]: done

File: python/foglamp/services/core/api/support.py (hunk 4 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/core/api/support.py]: done

File: python/foglamp/services/core/api/support.py (hunk 1 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/core/api/support.py]: done

**Comment by amarendra-dianomic** [comment]
Closed in favour of #650

**Pull request closed without merging**